ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/package.scala

```diff
@@ -100,25 +100,39 @@ package object v1 {
     * though that there usually are few elements in the array.
     *
     */
-  @SuppressWarnings(Array("org.wartremover.warts.ArrayEquals"))
-  final case class Offset(private val xs: Array[Int]) {
+  final case class Offset(private val xs: Array[Long]) {
     override def toString: String =
-      xs.mkString("-")
+      components.mkString("-")
 
-    def components: Iterable[Int] = xs
+    def components: Iterable[Long] = xs
+
+    override def equals(that: Any): Boolean = that match {
+      case o: Offset => Offset.compare(this, o) == 0
+      case _ => false
+    }
   }
   implicit object Offset extends Ordering[Offset] {
 
     /** Create an offset from a string of form 1-2-3. Throws
       * NumberFormatException on misformatted strings.
       */
     def assertFromString(s: String): Offset =
-      Offset(s.split('-').map(_.toInt))
+      Offset(s.split('-').map(_.toLong))
 
     override def compare(x: Offset, y: Offset): Int =
-      scala.math.Ordering.Iterable[Int].compare(x.xs, y.xs)
+      scala.math.Ordering.Iterable[Long].compare(x.xs, y.xs)
   }
 
+  /** The initial conditions of the ledger.
+    *
+    * @param ledgerId: The static ledger identifier.
+    * @param recordTimeEpoch: The initial record time before any [[Update]] arrives.
+    */
+  final case class LedgerInitialConditions(
+      ledgerId: LedgerId,
+      recordTimeEpoch: Timestamp
```

Comment:
Why is this called the `recordTimeEpoch`? Doesn't seem to match the the description.